src/Microsoft.ML.Data/Transforms/ValueToKeyMappingTransformer.cs

```diff
@@ -107,11 +107,11 @@ public abstract class ArgumentsBase : TransformInputBase
             [Argument(ArgumentType.AtMostOnce, HelpText = "Maximum number of terms to keep per column when auto-training", ShortName = "max", SortOrder = 5)]
             public int MaxNumTerms = ValueToKeyMappingEstimator.Defaults.MaxNumTerms;
 
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Comma separated list of terms", SortOrder = 105, Visibility = ArgumentAttribute.VisibilityType.CmdLineOnly)]
-            public string Terms;
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Comma separated list of terms", Name = "Terms", SortOrder = 105, Visibility = ArgumentAttribute.VisibilityType.CmdLineOnly)]
+            public string TermsList;
```

Comment:
>public string TermsList; [](start = 12, length = 24)

This looks  rather odd.  We are using suffix `List`  but its type is a `string`

Reply:
i think that is an overload for the below. 
Does it make sense to log an issue about making things like this internal? TextLoader has a few of those too. 

---
In reply to: [247227162](https://github.com/dotnet/machinelearning/pull/2124#discussion_r247227162) [](ancestors = 247227162)

Reply:
I opened this: [https://github.com/dotnet/machinelearning/issues/2133](https://github.com/dotnet/machinelearning/issues/2133)

---
In reply to: [247279013](https://github.com/dotnet/machinelearning/pull/2124#discussion_r247279013) [](ancestors = 247279013,247227162)

Reply:
It does look odd. And its not an overload for the one below. They are named `string Terms` and `string[] Term` originally :/. It complains if I name the string array "Terms" as well.

Also, it is breaking some command line tests, which complain about not finding "Terms" despite me adding `Name = "Terms"` to `string TermsList` in my branch. Reverting the names fixes it. Would you suggest not touching these in this PR and taking it up in #2133 ?

cc: @TomFinley

Reply:
Alternatively, I could rename `string[] Term` to `string[] TermArray` and keep `string Terms`, and decide about making one internal in @2133. Thoughts?